packages/components/src/DateTimePickers/InputDateTimeRangePicker/InputDateTimeRangePicker.component.js

```diff
@@ -26,6 +27,44 @@ function InputDateTimeRangePicker(props) {
 	const { id, dateFormat, useSeconds, onChange, onBlur } = props;
 	const inputProps = omit(props, PROPS_TO_OMIT_FOR_INPUT);
 
+	const [vertical, setVertical] = useState(false);
+	const containerRef = useRef();
+
+	const className = vertical
+		? classnames(theme['range-picker-vertical'], 'range-picker-vertical')
+		: classnames(theme['range-picker'], 'range-picker');
+
+	function tryHorizontal() {
+		const rangeContainer = containerRef.current;
+		if (rangeContainer) {
+			const nodes = rangeContainer.querySelectorAll('input');
+			const lastChild = nodes[nodes.length - 1];
+			if (lastChild) {
+				if (
+					rangeContainer.getBoundingClientRect().right < lastChild.getBoundingClientRect().right
+				) {
+					setVertical(true);
+				}
+			}
+		} else {
+			setVertical(false);
+		}
+	}
+
+	function showHorizontalAndTest() {
+		setVertical(false);
+		tryHorizontal();
+	}
```

Comment:
You can try to simplify those, perhaps have 1 function.
The test can be `rangeContainer.scrollWidth > rangeContainer.offsetWidth`, it's sufficient to know if its content overflows, instead of requesting the children.
You can merge those 2 function
```javascript
if (vertical) {
    setVertical(false);
}
// do the check and change set the new value if necessary
```